frontend/packages/operator-lifecycle-manager/src/components/catalog-source.tsx

```diff
@@ -112,77 +111,145 @@ const DefaultSourceKebab: React.FC<DefaultSourceKebabProps> = ({
+              label="Status"
+              obj={catalogSource}
+              path="status.connectionState.lastObservedState"
+            />
+            <DetailsItem label="Display Name" obj={catalogSource} path="spec.displayName" />
+            <DetailsItem label="Publisher" obj={catalogSource} path="spec.publisher" />
+            <DetailsItem label="Availability" obj={catalogSource} path="metadata.namespace">
+              {catsrcNamespace}
+            </DetailsItem>
+            <DetailsItem label="Endpoint" obj={catalogSource} path="spec.image">
+              {getEndpoint(catalogSource)}
+            </DetailsItem>
+            <DetailsItem
+              label="Registry Poll Interval"
+              obj={catalogSource}
+              path="spec.updateStrategy.registryPoll.interval"
+              canEdit={!_.isEmpty(catalogSource.spec.updateStrategy)}
+              onEdit={() => editRegitryPollInterval({ catalogSource })}
+            />
+            <DetailsItem label="Number of Operators" obj={catalogSource} path="">
+              {operatorCount}
+            </DetailsItem>
+          </div>
         </div>
       </div>
-      <div className="co-m-pane__body">
-        <SectionHeading text={t('catalog-source~Packages')} />
-        <PackageManifestList
-          loaded
-          data={packageManifests}
-          operatorGroup={toData(operatorGroups)}
-          subscription={toData(subscriptions)}
-        />
-      </div>
     </div>
   ) : (
     <div />
   );
 };
 
-export const CatalogSourceDetailsPage: React.FC<CatalogSourceDetailsPageProps> = (props) => (
-  <DetailsPage
-    {...props}
-    namespace={props.match.params.ns}
-    kind={referenceForModel(CatalogSourceModel)}
-    name={props.match.params.name}
-    pages={[navFactory.details(CatalogSourceDetails), navFactory.editYaml()]}
-    menuActions={Kebab.factory.common}
-    resources={[
-      {
-        kind: referenceForModel(PackageManifestModel),
-        isList: true,
-        namespace: props.match.params.ns,
-        selector: { matchLabels: { catalog: props.match.params.name } },
-        prop: 'packageManifests',
-      },
-      {
-        kind: referenceForModel(SubscriptionModel),
-        isList: true,
-        namespace: props.match.params.ns,
-        prop: 'subscriptions',
-      },
-      {
-        kind: referenceForModel(OperatorGroupModel),
-        isList: true,
-        namespace: props.match.params.ns,
-        prop: 'operatorGroups',
-      },
-    ]}
-  />
-);
+export const Operators: React.FC<PackageManifestPageProps> = (props) => {
+  return <PackageManifestsPage catalogSource={props.obj} showTitle={false} {...props} />;
+};
+
+export const CatalogSourceDetailsPage: React.FC<CatalogSourceDetailsPageProps> = (props) => {
+  const [operatorHub, operatorHubLoaded, operatorHubLoadError] = useK8sWatchResource<
+    OperatorHubKind
+  >({
+    kind: referenceForModel(OperatorHubModel),
+    name: 'cluster',
+    isList: false,
+  });
```

Comment:
Since the OperatorHub resource is a singular, unique cluster configuration resource, we could probably abstract this out into a custom reusable hook like `useOperatorHubConfig`. I'd put this in the `./packages/operator-lifecycle-manager/src/utils` folder with the only other custom olm hook.